src/elements/DipEdge.H

```diff
@@ -104,6 +138,34 @@ namespace impactx::elements
                 * m_g * m_K2 / powi<2>(m_rc);
 
             m_R43 = vf - m_R21;
+
+            // constants for nonlinear map:
+
+            // access reference particle values to find beta
+            m_beta = refpart.beta();
+
+            // expressions from eq (14) of Mitchell and Hwang, NAPAC2025
+            amrex::ParticleReal const tan_psi = sin_psi/cos_psi;
+            amrex::ParticleReal const sec_psi = 1_prt/cos_psi;
+
+            // the scale constant specifying entry or exit
+            amrex::ParticleReal const loc = static_cast<amrex::ParticleReal>(m_location);
```

Comment:
Oh fun, that cast returns now with `AMREX_ENUM` @WeiqunZhang 

```
loc=543450482 m_location=entry | 543450482
```

scratching my head why it worked with
```C++
enum Location
{
    entry = 1, //! for leading (entry) fringe field
    exit = -1  //! for trailing (exit) fringe field
};

```

Reply:
Interesting, works with a normal enum class of int well...
https://godbolt.org/z/3YY3bnT3h